### DISCUSSION TITLE
Feature Request: Dual account support/troubles

### DISCUSSION THREAD

**Discussion by mazhigbee-pb** [community/community#13] [Mobile]
I have 2 GitHub accounts that I use, I'd love to be able to switch between them both in the profile section of the app or something. 

Also since the app uses a web view to authenticate if I log into the app once and then directly sign out. When I go to sign back in again my session is valid in the web view on the same account (which is nice) but maybe there should be a "Not you?" link on the oauth screen so I can signout there. I found it quite tricky to clear that webview of my session. 

Keep up the great work, loving everything so far!

**Comment by eliperkins** [comment]
Hey @mazhigbee-pb! We hear ya. Account switching has been a very popular request from folks. We've got some ideas on how to tackle this in the future. Stay tuned!

> Also since the app uses a web view to authenticate if I log into the app once and then directly sign out. When I go to sign back in again my session is valid in the web view on the same account (which is nice) but maybe there should be a "Not you?" link on the oauth screen so I can signout there. I found it quite tricky to clear that webview of my session.

We've heard feedback on this as well! A future update of GitHub for iOS will no longer re-use previous sessions to log back in. I'll keep you posted!

**Reply by suprstarrd** [reply]
Checking in a few years later; how would this even work? GitHub's Terms of Service lays it out:
> * [...] You may maintain no more than one free machine account in addition to your free User Account.
> * One person or legal entity may maintain no more than one free Account (if you choose to control a machine account as well, that's fine, but it can only be used for running a machine).

So that's a limit of two accounts, and only one can actually be used? How does account switching fall under this?
